p2p/transport/webtransport/transport.go

```diff
@@ -207,10 +231,28 @@ func (t *transport) dial(ctx context.Context, addr ma.Multiaddr, url, sni string
 			return verifyRawCerts(rawCerts, certHashes)
 		}
 	}
-	conn, err := t.connManager.DialQUIC(ctx, addr, tlsConf, t.allowWindowIncrease)
-	if err != nil {
-		return nil, nil, err
+
+	var conn quic.Connection
+	var err error
+	if listener, err := t.findListenerForDial(addr); err == nil {
+		clientConf := t.connManager.ClientConfig().Clone()
+		clientConf.AllowConnectionWindowIncrease = t.allowWindowIncrease
+		naddr, _, err := quicreuse.FromQuicMultiaddr(addr)
+		if err != nil {
+			return nil, nil, err
+		}
+
+		conn, err = listener.reuseListener.Dial(ctx, naddr, tlsConf, clientConf)
+		if err != nil {
+			return nil, nil, err
+		}
```

Comment:
Should we expose this as a method in `ConnManager` as `DialWithListener` or `DialWithTransport` or something. I am specifically interested in keeping this logic in one place. 

```
clientConf := t.connManager.ClientConfig().Clone()
clientConf.AllowConnectionWindowIncrease = t.allowWindowIncrease
naddr, _, err := quicreuse.FromQuicMultiaddr(addr)
if err != nil {
	return nil, nil, err
}
```

Reply:
It would be a small benefit for reducing reuse here, but at the expense of introducing a new public method in ConnManager. If ConnManager was internal, I think that would be okay. But I don't think the reuse here is worth the tradeoff.